sdk/core/azure-core/test/ut/datetime.cpp

```diff
@@ -9,49 +9,54 @@
 
 using namespace Azure::Core;
 
+namespace {
+static const auto year1601 = DateTime(1601);
+static const auto year1801 = DateTime(1801);
+static const auto year2001 = DateTime(2001);
+static const auto zeroDuration = DateTime::Duration();
+} // namespace
+
 TEST(DateTime, ParseDateAndTimeBasic)
 {
-  auto dt1 = DateTime::FromString("20130517T00:00:00Z", DateTime::DateFormat::Iso8601);
-  EXPECT_NE(0u, dt1.ToInterval());
-
-  auto dt2 = DateTime::FromString("Fri, 17 May 2013 00:00:00 GMT", DateTime::DateFormat::Rfc1123);
-  EXPECT_NE(0u, dt2.ToInterval());
+  auto dt1 = DateTime::Parse("20130517T00:00:00Z", DateTime::DateFormat::Rfc3339);
```

Comment:
If I pass in Rfc1123 here, will it throw? And same below? Worth adding that as a test (mix-match format and input string).